Userland/uniq.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, Matthew L. Curry <matthew.curry@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/QuickSort.h>
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio", nullptr) > 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    UNUSED_PARAM(argc);
+    UNUSED_PARAM(argv);
```

Comment:
The uniq utility should be able to take input and output file paths as arguments, among other things (but I wouldn't focus on the other switches for now)

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/uniq.html

Reply:
I agree, I only avoided it in this initial version to get something that works alongside sort, which also does not take files yet. Should I go ahead and work on that?

Reply:
There are clearly improvements I can make here, I'll bring more soon.

Reply:
I have included positional arguments for input and output files according to The Open Group spec, along with the appropriate pledge. I've also cleaned up header files to just include the ones I'm really using.

Reply:
Looks good to me